### PR TITLE
Clean up references table

### PR DESCRIPTION
**Pull request by jmelot**
Closes https://github.com/georgetown-cset/cset_article_schema/issues/51

I've made sure the dag loads successfully but have not run this

### CODE REVIEW

File: schemas/references.json [hunk 1]
```diff
@@ -11,4 +11,4 @@
     "type": "STRING",
     "description": "Articles referenced by `id`"
```
Comment (jamesdunham) [inline schemas/references.json:12]: Now by `merged_id`

File: sql/references.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
+-- Create a table giving each publication's references, identified by their merged_ids.
+-- The all_metadata_with_cld2_lid table contains references identified by vendor IDs for each orig_id.
+-- For each merged_id, we take all its orig_ids' references, and look up the merged_ids of the references.
+-- We exclude references that appear outside our merged corpus.
+SELECT
+  DISTINCT links1.merged_id AS merged_id,
+  links2.merged_id AS ref_id
+FROM (
```
Comment (jamesdunham) [inline sql/references.sql:8]: I think we'd prefer a CTE here over a subquery under the style guide

File: sql/references.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  CROSS JOIN
+    UNNEST(SPLIT(references, ",")) AS reference
+  WHERE
+    reference IN (
+    SELECT
+      orig_id
+    FROM
+      {{ staging_dataset }}.article_links_with_dataset )) AS references
+LEFT JOIN
+  {{ staging_dataset }}.article_links AS links1
```
Comment (jamesdunham) [inline sql/references.sql:23]: It'd be clearer to alias these joins something other than `links1` and `links2`. IIUC one gets the merged IDs for our referencing papers and the other gets the merged IDs for our referenced papers. How about calling them something like `referencing_papers` or `papers` and then `referenced_papers`?

**Review by jamesdunham — APPROVED** [review]
Look correct to me. Suggesting minor changes for clarity.

**Comment by jmelot** [comment]
Good comments, ty!

**Pull request merged**